src/Tools/ExternalAccess/Extensions/Internal/ExtensionDocumentMessageHandlerWrapper.cs

```diff
@@ -0,0 +1,54 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Reflection;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.CodeAnalysis.Extensions;
+
+internal sealed class ExtensionDocumentMessageHandlerWrapper : IExtensionDocumentMessageHandlerWrapper
```

Comment:
so why we need the interface type, instead of just having this concrete type that everyone can refer to?

Reply:
tbh... not even sure why we have the type at all.

Reply:
@dibarbet didn't want other Roslyn projects to depend on the ExternalAccess project. So they can't "see" the `ExtensionDocumentMessageHandlerWrapper` type or the `IExtensionDocumentMessageHandler` interface.

Other minor advantage of the wrapper class is to hide the reflection needed to invoke a generic method.

Reply:
Right - usually the EA layer should be one directional (it references other Roslyn projects).  Otherwise various protocol projects have to reference the EA layer itself, which creates weird circuitous references